src/components/ProductFilterModal.vue

```diff
@@ -84,26 +92,47 @@ export default defineComponent({
       modalController.dismiss({ dismissed: true, isFilterChanged: this.isFilterChanged });
     },
     async search(event: any) {
-      // TODO: need to implement infinite scroll on the modal search
+      this.queryString = event.target.value;
+      this.facetOptions = [];
+      this.getTags();
+    },
+    async getTags(vSize?: any, vIndex?: any) {
+      const viewSize = vSize ? vSize : process.env.VUE_APP_VIEW_SIZE;
+      const viewIndex = vIndex ? vIndex : 0;
+
       const payload = {
         facetToSelect: this.facetToSelect,
         docType: 'PRODUCT',
         coreName: 'enterpriseSearch',
         searchfield: this.searchfield,
         jsonQuery: '{"query":"*:*","filter":["docType:PRODUCT"]}',
         noConditionFind: 'N',
-        limit: 50,
-        q: event.target.value,
-        term: event.target.value
+        limit: viewSize,
+        q: this.queryString,
+        term: this.queryString,
+        offset: viewIndex * viewSize,
       }
 
       const resp = await ProductService.fetchFacets(payload);
       if (resp.status == 200 && resp.data.length > 0) {
-        this.facetOptions = resp.data.map((obj: any) => ({ id: obj.id, label: obj.label }))
+        if(!this.facetOptions.length) {
+          this.facetOptions = resp.data.map((obj: any) => ({ id: obj.id, label: obj.label }))
+        } else {
+          this.facetOptions.push(...resp.data.map((obj: any) => ({ id: obj.id, label: obj.label })))
+          this.isScrollable = !(this.facetOptions.length < process.env.VUE_APP_VIEW_SIZE)
+        }
       } else {
-        this.facetOptions = []
```

Comment:
I think we should not remove this. When we have items in list for existing query, and subsequent query returns no results. It will give wrong results

Reply:
Yes sir. Added it back.